types/index.d.ts

```diff
@@ -96,6 +96,15 @@ interface EncoderOptions {
   syncMarker: Buffer;
 }
 
+/**
+ * A projection function that is used when unwrapping unions.
+ * This function is called at schema parsing time on each union with its branches' types.
+ * If it returns a non-null (function) value, that function will be called each time a value's branch needs to be inferred and should return the branch's index.
+ * The index muss be a number between 0 and length-1 of the passed types.
+ * In this case (a branch index) the union will use an unwrapped representation. Otherwise (undefined), the union will be wrapped.
+ */
+type ProjectionFn = (types: ReadonlyArray<Type>) => ((val: unknown) => number) | undefined;
```

Comment:
Can we rename to `BranchProjection`? Slightly more explicit and avoids introducing a new abbreviation.

Please also reformat the comment to fit within 80 characters.

Reply:
040aed7

Reply:
040aed7